### PR TITLE
Place current release at the top of version dropdown in Manage Packages dialog.

### LINKED ISSUE #12792: Kqlmagic in Azure Data Studio should behave like pip install where it picks the latest best non-dev version
- Azure Data Studio Version:
Version: 1.22.1
Commit: 1b5c54dd8c6eb03bf7ed1e04889d9f41de0523b3
Date: 2020-09-29T16:38:23.291Z
VS Code: 1.49.0
Electron: 9.2.1
Chrome: 83.0.4103.122
Node.js: 12.14.1
V8: 8.3.110.13-electron.0
OS: Linux x64 5.4.0-48-generic

Steps to Reproduce:

1. Use **Manage Packages** on a Python Notebook. 
2. Click on **Add New** then search for Kqlmagic. 
Actual - 
image

Expected - 
pip upgrade command today installs/upgrades kqlmagic with the latest non-dev version. It would be best to match this behavior. 
i.e. only list 0.1.11, 0.1.12, 0.1.13, 0.1.113post etc. 
(For testing purposes, users can install the dev versions from pip manually if needed)

### PR DESCRIPTION
**Pull request by corivera**
This PR fixes #12792

Also updated version comparison to better handle non-numeric version components. The new behavior falls back to normal string comparison if either version number isn't actually a number. Like comparing "1.1.**0**" and "1.1.**0a**", for example.

Fun fact: parseInt() will parse integers from a string until it hits a non-number character. So "1a" resolves to the number 1, but "a1a" resolves to NaN. Using the Number wrapper returns NaN in both of those cases, as expected.

### CODE REVIEW

**Comment by Charles-Gagnon** [comment]
`So "1a" resolves to the number 1, but "a1a" resolves to NaN`

https://www.destroyallsoftware.com/talks/wat

File: extensions/notebook/src/common/utils.ts [hunk 1]
```diff
@@ -144,22 +144,31 @@ export function getOSPlatformId(): string {
  * @returns 1 if the first version is greater, -1 if it's less, and 0 otherwise.
  */
 export function comparePackageVersions(first: string, second: string): number {
-	let firstVersion = first.split('.').map(numStr => Number.parseInt(numStr));
-	let secondVersion = second.split('.').map(numStr => Number.parseInt(numStr));
+	let firstVersion = first.split('.');
```
Comment (Charles-Gagnon) [inline extensions/notebook/src/common/utils.ts:147]: Why not use a package like semver instead? We do that in most other places we need to compare versions.

**Review by Charles-Gagnon — APPROVED** [review]

File: extensions/notebook/src/common/utils.ts (hunk 1 above)
Comment (corivera) [reply extensions/notebook/src/common/utils.ts:147]: Semver only takes exactly 3 version numbers, from what I remember. I remember using it way back when we first added this, but there were some package cases that semver errored on.

Comment (corivera) [reply extensions/notebook/src/common/utils.ts:147]: Looks like semver.coerce() can convert a version string into a valid semver format, but then you lose information. From the semver usage docs:
`semver.valid(semver.coerce('v2')) // '2.0.0'`

`semver.valid(semver.coerce('42.6.7.9.3-alpha')) // '42.6.7'`

Comment (corivera) [reply extensions/notebook/src/common/utils.ts:147]: https://www.npmjs.com/package/semver#usage

**Pull request merged**